src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -281,9 +282,19 @@ RadixContainer<T> materialize_input(const std::shared_ptr<const Table>& in_table
       auto reference_chunk_offset = ChunkOffset{0};
 
       const auto segment = chunk_in->get_segment(column_id);
-      segment_with_iterators<T>(*segment, [&](auto it, const auto end) {
+      segment_with_iterators<T>(*segment, [&](auto it, auto end) {
         using IterableType = typename decltype(it)::IterableType;
 
+        if (dynamic_cast<ValueSegment<T>*>(&*segment)) {
```

Comment:
Just curious: why not `std::dynamic_pointer_cast`?

Reply:
Because it would copy-construct the shared_ptr, incrementing and decrementing the reference count for a variable that we do not use.

`typeid(...) == ...` would be another option, but I think it doesn't make a difference compared to a regular dynamic cast.